src/core.js

```diff
@@ -1071,7 +1058,7 @@ define([
                 _this.data.core.formLoadingFailed = true;
                 _this.data.core.failedLoadXML = formString;
 
-                $.fancybox.hideActivity();
+                _this.closeModal();
```

Comment:
This is going to close the modal that pops up telling the user they have an error.

I think the best solution may be to modify closeModal to conditionally take in an argument that will close a specific modal. You may need to add unique ids to each modal though.